compiler/driver.ml

```diff
@@ -29,66 +29,89 @@ let modname_of_file f =
 let load_module_interfaces options includes program =
   (* Recurse into program modules, looking up files in [using] and loading
      them *)
+  if program.Surface.Ast.program_used_modules <> [] then
+    Message.emit_debug "Loading module interfaces...";
   let includes =
     includes
     |> List.map (fun d -> File.Tree.build (options.Cli.path_rewrite d))
     |> List.fold_left File.Tree.union File.Tree.empty
   in
   let err_req_pos chain =
-    List.map (fun m -> Some "Module required from", ModuleName.pos m) chain
+    List.map (fun mpos -> Some "Module required from", mpos) chain
   in
-  let find_module req_chain m =
-    let fname_base = ModuleName.to_string m in
-    let required_from_file = Pos.get_file (ModuleName.pos m) in
+  let find_module req_chain (mname, mpos) =
+    let required_from_file = Pos.get_file mpos in
     let includes =
       File.Tree.union includes
         (File.Tree.build (File.dirname required_from_file))
     in
     match
       List.filter_map
-        (fun (ext, _) -> File.Tree.lookup includes (fname_base ^ ext))
+        (fun (ext, _) -> File.Tree.lookup includes (mname ^ ext))
         extensions
     with
     | [] ->
       Message.raise_multispanned_error
-        (err_req_pos (m :: req_chain))
-        "Required module not found: %a" ModuleName.format m
+        (err_req_pos (mpos :: req_chain))
+        "Required module not found: @{<blue>%s@}" mname
     | [f] -> f
     | ms ->
       Message.raise_multispanned_error
-        (err_req_pos (m :: req_chain))
-        "Required module %a matches multiple files: %a" ModuleName.format m
+        (err_req_pos (mpos :: req_chain))
+        "Required module @{<blue>%s@} matches multiple files:@;<1 2>%a" mname
         (Format.pp_print_list ~pp_sep:Format.pp_print_space File.format)
         ms
   in
-  let load_file f =
-    let (mname, intf), using =
-      Surface.Parser_driver.load_interface (Cli.FileName f)
-    in
-    (ModuleName.of_string mname, intf), using
-  in
-  let rec aux req_chain acc modules =
+  (* modulename * program * (id -> modulename) *)
```

Comment:
you can turn that into a proper type annotation no?

Reply:
Takes 5x as much space but fine ;)